src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -534,29 +531,39 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
                 }
             })?;
 
-        // FIXME: are these both necessary?
-        let ty_res = if let Some(ty_res) = resolve_primitive(&path_root, TypeNS)
```

Comment:
I'm bisecting this now - I think it's unlikely this is the cause, but it's possible.

Reply:
Oh! I figured it out - `resolve_path` gives precedence to modules over primitives, this does the reverse.
```
|      ^^^^^^^^^ no item named `MAX` in module `usize`
```
The issue is that both `mod usize` and the primitive usize are in scope. I think this might actually be a bug even before - rustdoc should give an error if the base res is ambiguous, but it doesn't:

```
$ cat primitive-ambiguous.rs 
//! [usize::MAX]

mod usize {
  const MAX: usize = 1;
}
$ rustdoc primitive-ambiguous.rs
$
```

For this specific test case, I think we could fix it by not giving an ambiguity error if the module has `doc(primitive)` on it?

Reply:
I don't think I'm qualified to have a say in this :)

Reply:
cc @Manishearth - do you have an opinion here? For context, the original test case was
```rust
//! [usize::MAX]

#[doc(primitive = "usize")]
mod usize {}
```

Reply:
Actually I think I'm going to wait to fix this for a follow-up PR, it's a change behavior and shouldn't be mixed with the cleanups.